src/crewai/llm.py

```diff
@@ -568,6 +593,52 @@ def _handle_streaming_response(
             )
             raise Exception(f"Failed to get streaming response: {str(e)}")
 
+    def _handle_streaming_tool_calls(
+        self,
+        tool_calls: List[Any],
+        accumulated_tool_args: DefaultDict[int, SimpleNamespace],
+        available_functions: Optional[Dict[str, Any]] = None,
+    ) -> None | str:
+        for tool_call in tool_calls:
+            if tool_call.function.name:
+                accumulated_tool_args[tool_call.index].function.name = (
+                    tool_call.function.name
+                )
+
+            if tool_call.function.arguments:
+                accumulated_tool_args[
+                    tool_call.index
+                ].function.arguments += tool_call.function.arguments
+
+            crewai_event_bus.emit(
+                self,
+                event=LLMStreamChunkEvent(
+                    tool_call=tool_call.to_dict(),
+                    chunk=tool_call.function.arguments,
+                ),
+            )
+
+            if (
+                accumulated_tool_args[tool_call.index].function.name
+                and accumulated_tool_args[tool_call.index].function.arguments
+                and available_functions
+            ):
+                try:
+                    # Try to parse the accumulated arguments
+                    json.loads(
+                        accumulated_tool_args[tool_call.index].function.arguments
+                    )
+
+                    # Execute the tool call
+                    return self._handle_tool_call(
+                        [accumulated_tool_args[tool_call.index]],
+                        available_functions,
+                    )
+                except json.JSONDecodeError:
+                    # If JSON is incomplete, continue accumulating
+                    continue
```

Comment:
I think your code is quite self explanatory, so we can remove comments if you agree.